### DISCUSSION TITLE
fetch all the data using findFirst

### DISCUSSION THREAD

**Discussion by rajamoulimallareddy** [prisma/prisma#16468] [Q&A] [answered]
```
model test {
  id   Int    @id @default(autoincrement())
  name String
  tt2  tet2[]
}

model tet2 {
  id     Int    @id @default(autoincrement())
  n2     String
  test   test?  @relation(fields: [testId], references: [id])
  testId Int?
}
```

how do i fetch in typescript using findfirst

i wanted fetch the data of both from `test` how do i do it?

**Comment by nurul3101** [comment]
Hello @rajamoulimallareddy 👋 

If you want to fetch tet2 data along with test data you can use the [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) clause.
Here's an example script:
```typescript
  await prisma.test.findFirst({
    where: {
      name: 'test',
    },
    include: {
      tt2: true,
    },
  });
```

Let me know if this is what you needed or in case you have any further questions.

**Comment by rajamoulimallareddy** [comment, marked as answer]
Hey thanks 🙏. One more doubt. If it is kept true the whole data will be shown? Or should be define more keys in that

**Reply by nurul3101** [reply]
It would show all the fields of `tet2` model, if you want to select some specific fields then you can do something like this:
```ts
await prisma.test.findFirst({
    where: {
      name: 'test',
    },
    include: {
      tt2: {
        select: {
          id: true,
          n2: true,
        },
      },
    },
  });
```

**Reply by rajamoulimallareddy** [reply]
Okay and also one more thing. In tt2 we kept tet2 inside an array right instead cant we leave it as a object?

**Reply by nurul3101** [reply]
the [] besides tet2 indicates that there is one-to-many relation between test and tet2 models. If there is a one-to-one relation then you can keep it like this as well:
```
model test {
  id   Int    @id @default(autoincrement())
  name String
  tt2  tet2?
}

model tet2 {
  id     Int    @id @default(autoincrement())
  n2     String
  test   test?  @relation(fields: [testId], references: [id])
  testId Int?   @unique
}
```
You can learn more about relations in this [guide](https://www.prisma.io/docs/concepts/components/prisma-schema/relations)

**Reply by nurul3101** [reply]
Can you open a new GitHub Discussion for it? Other users who might be running into this same issue might not be able to find it in this Discussion 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
